Name: Launcher: Reset to Middle Page
Version: 1.3.1-27
Author: Ian Grant
Description: Resets the launcher when closing so that it always opens to the middle page, rather than the last page used. Obviously this works best with an odd number of pages (like the default 3). It also scrolls each page back to the top.  (MIT license)

--- /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js.orig	Sat Dec  5 11:48:12 2009
+++ /usr/lib/luna/system/luna-applauncher/app/controllers/launcher-assistant.js	Thu Dec  3 19:42:03 2009
@@ -183,6 +183,25 @@
 		delete this.launchRequest;
 		
 		this.removeLaunchFeedback();
+	
+		/**
+		 * Calculate middle page. For an even number of pages,
+		 * this will be the lower side of the mid-point.
+		 */
+		var numPages=this.pagesModel.getNumPages();
+		var middlePage = (numPages % 2) ? (numPages - 1) / 2: numPages / 2 - 1;
+		
+		/* Reset launcher to middle page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo((middlePage) * -this.kPageWidth, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(middlePage, false);
+		this.updatePageIndicators();
+		
+		/* Jump scroll position back to top for all pages */
+		for (var i=0; i<numPages; i++) {
+			var scroller = this.getPageScroller(i);
+			if (scroller && scroller.mojo) { scroller.mojo.revealTop(); }
+		}
 	},
 	
 	onResize: function(event) {
@@ -529,6 +545,18 @@
 		}
 		
 		this.globalSearchAssistant.enable(true);
+		
+		/**
+		 * Calculate middle page. For an even number of pages,
+		 * this will be the lower side of the mid-point.
+		 */
+		var middlePage = (numPages % 2) ? (numPages - 1) / 2: numPages / 2 - 1;
+		
+		/* Set launcher to middle page - seems we need to do scrollTo AND setSnapIndex to be fully compatible with Wrap Pages */
+		$('launcher_root').mojo.scrollTo((middlePage) * -this.kPageWidth, undefined, false, true);
+		this.activePageIndex = Math.abs($('launcher_root').mojo.getScrollPosition().left) / this.kPageWidth;
+		$('launcher_root').mojo.setSnapIndex(middlePage, false);
+		this.updatePageIndicators();
 	},
 	
 	/* re-acquires the set of page elements the scroller needs to snap */
